### PR TITLE
Ex5

### PR DESCRIPTION
**Pull request by zyahav**

### CODE REVIEW

File: ex1/ex1.js [hunk 1]
```diff
@@ -6,3 +6,29 @@
  *     fn(",,,") === 2
```
Comment (myok12) [inline ex1/ex1.js]: Please undo all files except ex5.js on this branch

File: ex5/ex5Spec.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+describe("firstUniqueChar", function() {
+	it("finds 'e' as the first character in the string", function() {
+		expect(ex5.firstUniqueChar("abcdedcba")).toBe("e");
+	});
+
+	it("return null, we do not have a uniq char on string", function() {
+		expect(ex5.firstUniqueChar("abcdeedcba")).toBeNull();
+	});
+
+	it("finds 'c' as the first character in the string", function() {
```
Comment (myok12) [inline ex5/ex5Spec.js]: The 'c' isn't that important here. Just change the it to: finds another unique character in the string.

File: ex5/ex5Spec.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		expect(ex5.firstUniqueChar("abababcab")).toBe("c");
+	});
+
+	it("return null, all chars in string are same", function() {
+		expect(ex5.firstUniqueChar("aaaaaaaaa")).toBeNull();
+	});
+
+	it("finds 'a' as the first character in the strin", function() {
```
Comment (myok12) [inline ex5/ex5Spec.js]: rename test to: works for a single character.

File: ex5/ex5Spec.js (hunk 2 above)
Comment (myok12) [inline ex5/ex5Spec.js]: Same for the 'e' above.

File: ex5/ex5.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * firstUniqueChar("a") === "a"
+ */
+
+function firstUniqueChar(str) {
+
+	var doNotLookForCharList = '', searchChar = '', searchIndex = 0; 
+
+	for (searchIndex in str) {
+
+		searchIndex = parseInt(searchIndex);
```
Comment (myok12) [inline ex5/ex5.js]: Why do you need to parseInt?

File: ex5/ex5.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * firstUniqueChar("abababcab") === "c"
+ * firstUniqueChar("aaaaaaaaa") === null
+ * firstUniqueChar("a") === "a"
+ */
+
+function firstUniqueChar(str) {
+
+	var doNotLookForCharList = '', searchChar = '', searchIndex = 0; 
+
+	for (searchIndex in str) {
```
Comment (myok12) [inline ex5/ex5.js]: Are you looping over indexes? If so, switch to a regular for loop: `for (var searchIndex = 0; searchIndex<str.length; searchIndex++)`

File: ex5/ex5.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+		searchIndex = parseInt(searchIndex);
+
+		searchChar = str.charAt(searchIndex);
+
+		//  Let's see if we did look for searchChar
+		if (doNotLookForCharList.includes(searchChar) != true) {
```
Comment (myok12) [inline ex5/ex5.js]: includes is not a common API. You can either leave this as is, or switch to `.indexOf() !== -1`.

If you leave as is, please change to: `if (!doNotLookForCharList.includes(searchChar)) {`

File: ex5/ex5.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+			//  SearchChar is not in doNotLookForCharList so will search for it
+			if (str.indexOf(searchChar, searchIndex + 1) < 0) {
```
Comment (myok12) [inline ex5/ex5.js]: It looks like you are searching ahead (with the indexOf) and before (using doNotLookForCharList). I don't think you need both methods.

File: ex5/ex5Spec.js (hunk 2 above)
Comment (zyahav) [inline ex5/ex5Spec.js]: done

File: ex5/ex5.js (hunk 5 above)
Comment (zyahav) [inline ex5/ex5.js]: done

File: ex5/ex5.js (hunk 4 above)
Comment (zyahav) [inline ex5/ex5.js]: now i dont 
searchIndex is a string type so need to number

File: ex5/ex5.js (hunk 6 above)
Comment (zyahav) [inline ex5/ex5.js]: you did not understand what i did
searchChar = str.charAt(searchIndex); >>> get the char value
and your .indexOf() returns index of char value

i will str[searchIndex]

File: ex5/ex5.js (hunk 7 above)
Comment (zyahav) [inline ex5/ex5.js]: if find char in doNotLookForCharList will move to next char so??????
  for (var searchIndex = 0; searchIndex < str.length; searchIndex++) {

File: ex5/ex5.js (hunk 6 above)
Comment (myok12) [inline ex5/ex5.js]: I am fine with the `charAt` call. I was asking you to replace the `includes`. Re-read what I wrote.

**Pull request closed without merging**